plugin/src/main/java/org/gradle/testretry/internal/executer/framework/BaseJunitTestFrameworkStrategy.java

```diff
@@ -118,7 +118,7 @@ private boolean processSpockTest(TestFilterBuilder filters, TestsReader testsRea
 
     private void addPotentiallyParameterizedSuffixed(TestFilterBuilder filters, String className, String name) {
         // It's a common pattern to add all the parameters on the end of a literal method name with []
-        String strippedParameterName = name.replaceAll("(?:\\([^)]*?\\)|\\[[^]]*?])*$", "");
+        String strippedParameterName = name.replaceAll("(?:\\([^)]*?\\))?(?:\\[[^]]*?](?:\\([^)]*?\\))?)?$", "");
```

Comment:
🤯

Reply:
+1, can we add a comment explaining what those things should match? Or maybe a unit test?

Reply:
I've realised I made the regex more complicated than necessary. I've updated it to match only `(...)`, or `(...)[...]`, added a comment, and an additional test case in `ParenthesesFuncTest`